test/testdrive/protobuf-repeated.td

```diff
@@ -0,0 +1,56 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that repeated fields in a Protobuf message are decoded correctly.
+
+$ file-append path=repeated.proto
+syntax = "proto3";
+
+message Repeated {
+    enum Enum {
+        ENUM0 = 0;
+        ENUM1 = 1;
+    }
+
+    message Message {
+        int32 m1 = 1;
+        int32 m2 = 2;
+    }
+
+    repeated bool bool = 1;
+    repeated Enum enum = 8;
+    repeated Message message = 9;
+}
+
+$ protobuf-compile-descriptors inputs=repeated.proto output=repeated.pb
+
+$ kafka-create-topic topic=repeated partitions=1
+
+$ kafka-ingest topic=repeated format=protobuf descriptor-file=repeated.pb message=.Repeated
+{"bool": [true, false], "enum": ["ENUM1", "ENUM0"], "message": [{"m1": 4, "m2": 2}, {"m1": 2, "m2": 4}]}
+{}
+
+> CREATE MATERIALIZED SOURCE repeated FROM
+  KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-repeated-${testdrive.seed}'
+  FORMAT PROTOBUF MESSAGE '.Repeated' USING SCHEMA FILE '${testdrive.temp-dir}/repeated.pb'
+
+# TODO(benesch): all of these columns should be non-nullable.
```

Comment:
This and similar errors in other files are still planned to be fixed by your #9019 yeah?

Reply:
Yep exactly!